File: Components/ASP.NET/VB/ASPxEditorsDemos/App_Code/EditorsSampleDbModel.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Data.Entity
Imports System.Data.Entity.Infrastructure
Imports System.Data.Entity.ModelConfiguration

Partial Public Class EditorsSampleDBContext
	Inherits ContextBase
	Shared Sub New()
		Database.SetInitializer(Of EditorsSampleDBContext)(Nothing)
	End Sub

	Public Sub New()
		MyBase.New("EditorsSampleDBConnectionString")
	End Sub

	Public Property ClubMembers() As DbSet(Of ClubMember)
	Public Property Occupations() As DbSet(Of Occupation)
	Public Property Tasks() As DbSet(Of Task)
	Public Property UserRoles() As DbSet(Of UserRole)
	Public Property Users() As DbSet(Of User)

	Protected Overrides Sub OnModelCreating(ByVal modelBuilder As DbModelBuilder)
		modelBuilder.Configurations.Add(New ClubMemberMap())
		modelBuilder.Configurations.Add(New OccupationMap())
		modelBuilder.Configurations.Add(New TaskMap())
		modelBuilder.Configurations.Add(New UserRoleMap())
		modelBuilder.Configurations.Add(New UserMap())
	End Sub
End Class

Partial Public Class ClubMember
	Public Property ClubMemberId() As Integer
	Public Property Name() As String
	Public Property Email() As String
	Public Property Birthday() As System.DateTime?
	Public Property Occupation() As String
	Public Property Gender() As String
	Public Property SubscriptionFlag() As Byte
End Class

Partial Public Class Occupation
	Public Property OccupationName() As String
End Class

Partial Public Class Task
	Public Property ID() As Integer
	Public Property Name() As String
	Public Property Priority() As String
	Public Property Status() As String
	Public Property Complete() As Integer?
	Public Property Assigned_To() As String
End Class
Partial Public Class User
	Public Property ID() As Integer
	Public Property UserName() As String
	Public Property Roles() As String
End Class

Partial Public Class UserRole
	Public Property ID() As Integer
	Public Property Name() As String
End Class

Public Class ClubMemberMap
	Inherits EntityTypeConfiguration(Of ClubMember)
	Public Sub New()
		' Primary Key
		Me.HasKey(Function(t) t.ClubMemberId)

		' Properties
		Me.Property(Function(t) t.Name).IsRequired().HasMaxLength(128)

		Me.Property(Function(t) t.Email).IsRequired().HasMaxLength(64)

		Me.Property(Function(t) t.Occupation).HasMaxLength(128)

		Me.Property(Function(t) t.Gender).HasMaxLength(6)

		' Table & Column Mappings
		Me.ToTable("ClubMember")
		Me.Property(Function(t) t.ClubMemberId).HasColumnName("ClubMemberId")
		Me.Property(Function(t) t.Name).HasColumnName("Name")
		Me.Property(Function(t) t.Email).HasColumnName("Email")
		Me.Property(Function(t) t.Birthday).HasColumnName("Birthday")
		Me.Property(Function(t) t.Occupation).HasColumnName("Occupation")
		Me.Property(Function(t) t.Gender).HasColumnName("Gender")
		Me.Property(Function(t) t.SubscriptionFlag).HasColumnName("SubscriptionFlag")
	End Sub
End Class

Public Class OccupationMap
	Inherits EntityTypeConfiguration(Of Occupation)
	Public Sub New()
		' Primary Key
		Me.HasKey(Function(t) t.OccupationName)

		' Properties
		Me.Property(Function(t) t.OccupationName).IsRequired().HasMaxLength(64)

		' Table & Column Mappings
		Me.ToTable("Occupation")
		Me.Property(Function(t) t.OccupationName).HasColumnName("Occupation")
	End Sub
End Class

Public Class TaskMap
	Inherits EntityTypeConfiguration(Of Task)
	Public Sub New()
		' Primary Key
		Me.HasKey(Function(t) t.ID)

		' Properties
		Me.Property(Function(t) t.Name).HasMaxLength(50)

		Me.Property(Function(t) t.Priority).HasMaxLength(50)

		Me.Property(Function(t) t.Status).HasMaxLength(50)

		Me.Property(Function(t) t.Assigned_To).HasMaxLength(50)

		' Table & Column Mappings
		Me.ToTable("Tasks")
		Me.Property(Function(t) t.ID).HasColumnName("ID")
		Me.Property(Function(t) t.Name).HasColumnName("Name")
		Me.Property(Function(t) t.Priority).HasColumnName("Priority")
		Me.Property(Function(t) t.Status).HasColumnName("Status")
		Me.Property(Function(t) t.Complete).HasColumnName("Complete")
		Me.Property(Function(t) t.Assigned_To).HasColumnName("Assigned To")
	End Sub
End Class

Public Class UserMap
	Inherits EntityTypeConfiguration(Of User)
	Public Sub New()
		' Primary Key
		Me.HasKey(Function(t) t.ID)

		' Properties
		Me.Property(Function(t) t.UserName).HasMaxLength(50)

		Me.Property(Function(t) t.Roles).HasMaxLength(255)

		' Table & Column Mappings
		Me.ToTable("Users")
		Me.Property(Function(t) t.ID).HasColumnName("ID")
		Me.Property(Function(t) t.UserName).HasColumnName("UserName")
		Me.Property(Function(t) t.Roles).HasColumnName("Roles")
	End Sub
End Class

Public Class UserRoleMap
	Inherits EntityTypeConfiguration(Of UserRole)
	Public Sub New()
		' Primary Key
		Me.HasKey(Function(t) t.ID)

		' Properties
		Me.Property(Function(t) t.Name).HasMaxLength(50)

		' Table & Column Mappings
		Me.ToTable("UserRoles")
		Me.Property(Function(t) t.ID).HasColumnName("ID")
		Me.Property(Function(t) t.Name).HasColumnName("Name")
	End Sub
End Class

File: Components/ASP.NET/CS/BootstrapDemos/App_Code/EditorsSampleDbModel.cs
using System;
using System.Data.Entity;
using System.Data.Entity.Infrastructure;
using System.Data.Entity.ModelConfiguration;

public partial class EditorsSampleDBContext : ContextBase {
    static EditorsSampleDBContext() {
        Database.SetInitializer<EditorsSampleDBContext>(null);
    }

    public EditorsSampleDBContext() : base("EditorsSampleDBConnectionString") { }

    public DbSet<ClubMember> ClubMembers { get; set; }
    public DbSet<Occupation> Occupations { get; set; }
    public DbSet<Task> Tasks { get; set; }
    public DbSet<UserRole> UserRoles { get; set; }
    public DbSet<User> Users { get; set; }

    protected override void OnModelCreating(DbModelBuilder modelBuilder) {
        modelBuilder.Configurations.Add(new ClubMemberMap());
        modelBuilder.Configurations.Add(new OccupationMap());
        modelBuilder.Configurations.Add(new TaskMap());
        modelBuilder.Configurations.Add(new UserRoleMap());
        modelBuilder.Configurations.Add(new UserMap());
    }
}

public partial class ClubMember {
    public int ClubMemberId { get; set; }
    public string Name { get; set; }
    public string Email { get; set; }
    public Nullable<System.DateTime> Birthday { get; set; }
    public string Occupation { get; set; }
    public string Gender { get; set; }
    public byte SubscriptionFlag { get; set; }
}

public partial class Occupation {
    public string OccupationName { get; set; }
}

public partial class Task {
    public int ID { get; set; }
    public string Name { get; set; }
    public string Priority { get; set; }
    public string Status { get; set; }
    public Nullable<int> Complete { get; set; }
    public string Assigned_To { get; set; }
}
public partial class User {
    public int ID { get; set; }
    public string UserName { get; set; }
    public string Roles { get; set; }
}

public partial class UserRole {
    public int ID { get; set; }
    public string Name { get; set; }
}

public class ClubMemberMap : EntityTypeConfiguration<ClubMember> {
    public ClubMemberMap() {
        // Primary Key
        this.HasKey(t => t.ClubMemberId);

        // Properties
        this.Property(t => t.Name)
            .IsRequired()
            .HasMaxLength(128);

        this.Property(t => t.Email)
            .IsRequired()
            .HasMaxLength(64);

        this.Property(t => t.Occupation)
            .HasMaxLength(128);

        this.Property(t => t.Gender)
            .HasMaxLength(6);

        // Table & Column Mappings
        this.ToTable("ClubMember");
        this.Property(t => t.ClubMemberId).HasColumnName("ClubMemberId");
        this.Property(t => t.Name).HasColumnName("Name");
        this.Property(t => t.Email).HasColumnName("Email");
        this.Property(t => t.Birthday).HasColumnName("Birthday");
        this.Property(t => t.Occupation).HasColumnName("Occupation");
        this.Property(t => t.Gender).HasColumnName("Gender");
        this.Property(t => t.SubscriptionFlag).HasColumnName("SubscriptionFlag");
    }
}

public class OccupationMap : EntityTypeConfiguration<Occupation> {
    public OccupationMap() {
        // Primary Key
        this.HasKey(t => t.OccupationName);

        // Properties
        this.Property(t => t.OccupationName)
            .IsRequired()
            .HasMaxLength(64);

        // Table & Column Mappings
        this.ToTable("Occupation");
        this.Property(t => t.OccupationName).HasColumnName("Occupation");
    }
}

public class TaskMap : EntityTypeConfiguration<Task> {
    public TaskMap() {
        // Primary Key
        this.HasKey(t => t.ID);

        // Properties
        this.Property(t => t.Name)
            .HasMaxLength(50);

        this.Property(t => t.Priority)
            .HasMaxLength(50);

        this.Property(t => t.Status)
            .HasMaxLength(50);

        this.Property(t => t.Assigned_To)
            .HasMaxLength(50);

        // Table & Column Mappings
        this.ToTable("Tasks");
        this.Property(t => t.ID).HasColumnName("ID");
        this.Property(t => t.Name).HasColumnName("Name");
        this.Property(t => t.Priority).HasColumnName("Priority");
        this.Property(t => t.Status).HasColumnName("Status");
        this.Property(t => t.Complete).HasColumnName("Complete");
        this.Property(t => t.Assigned_To).HasColumnName("Assigned To");
    }
}

public class UserMap : EntityTypeConfiguration<User> {
    public UserMap() {
        // Primary Key
        this.HasKey(t => t.ID);

        // Properties
        this.Property(t => t.UserName)
            .HasMaxLength(50);

        this.Property(t => t.Roles)
            .HasMaxLength(255);

        // Table & Column Mappings
        this.ToTable("Users");
        this.Property(t => t.ID).HasColumnName("ID");
        this.Property(t => t.UserName).HasColumnName("UserName");
        this.Property(t => t.Roles).HasColumnName("Roles");
    }
}

public class UserRoleMap : EntityTypeConfiguration<UserRole> {
    public UserRoleMap() {
        // Primary Key
        this.HasKey(t => t.ID);

        // Properties
        this.Property(t => t.Name)
            .HasMaxLength(50);

        // Table & Column Mappings
        this.ToTable("UserRoles");
        this.Property(t => t.ID).HasColumnName("ID");
        this.Property(t => t.Name).HasColumnName("Name");
    }
}


